src/cpp/fastdds/xtypes/dynamic_types/DynamicDataImpl.cpp

```diff
@@ -0,0 +1,7218 @@
+            }
+            else if (33 > sum)
+            {
+                current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+            }
+            else
+            {
+                current_alignment += 8 + eprosima::fastcdr::Cdr::alignment(current_alignment, 8);
+            }
+            break;
+        }
+        case TK_STRUCTURE:
+        {
+            if (ExtensibilityKind::FINAL != type_impl->get_descriptor().extensibility_kind())
+            {
+                // For APPENDABLE and MUTABLE, the maximum is the XCDR2 header (DHEADER(0) : Int32).
+                current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+            }
+
+            for (auto& member : type_impl->get_all_members_by_index())
+            {
+                auto member_impl = traits<DynamicTypeMember>::narrow<DynamicTypeMemberImpl>(member);
+
+                if (ExtensibilityKind::MUTABLE == type_impl->get_descriptor().extensibility_kind() ||
+                        member->get_descriptor().is_optional())
+                {
+                    // If member is from a MUTABLE type (or it is optional member) the maximum is XCDR1 LongMemberHeader.
+                    // << ALIGN(4)
+                    // << { FLAG_I + FLAG_M + PID_EXTENDED : UInt16 }
+                    // << { slength=8 : UInt16 }
+                    // << { M.id : <<: UInt32 }
+                    // << { M.value.ssize : UInt32 }
+                    current_alignment += 4 + 4 + 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+                }
+
+                current_alignment += calculate_max_serialized_size(
+                    member_impl->get_descriptor().type(), current_alignment);
+            }
+
+            if (ExtensibilityKind::MUTABLE != type_impl->get_descriptor().extensibility_kind())
+            {
+                // For MUTABLE, extra alignment for the PID_SENTINAL.
+                current_alignment += eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+            }
+            break;
+        }
+        case TK_ARRAY:
+        {
+            TypeKind element_kind =
+                    get_enclosing_typekind(traits<DynamicType>::narrow<DynamicTypeImpl>(
+                                type_impl->get_descriptor().element_type()));
+
+            if (is_complex_kind(element_kind) ||
+                    TK_STRING8 == element_kind ||
+                    TK_STRING16 == element_kind)
+            {
+                // DHEADER if XCDRv2
+                current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+            }
+
+            auto dimension {std::accumulate(type_impl->get_descriptor().bound().begin(),
+                                    type_impl->get_descriptor().bound().end(), 1, std::multiplies<uint32_t>())};
+
+            if (0 < dimension)
```

Comment:
this calculate the dimension, no the number of bounds. And the dimension could be 0 because it is valid in IDL
```omgidl
typedef char myarray[0];
```

Reply:
Done in c8f6dde98